app/V1Module/presenters/UsersPresenter.php

```diff
@@ -313,14 +313,26 @@ public function actionGroups(string $id) {
       throw new ForbiddenRequestException();
     }
 
+    list($archivedAsStudent, $asStudent) = $user->getGroupsAsStudent()->partition(function (Group $group) {
+      return $group->isArchived();
+    });
+
+    list($archivedAsSupervisor, $asSupervisor) = $user->getGroupsAsSupervisor()->partition(function (Group $group) {
+      return $group->isArchived();
+    });
+
     $this->sendSuccessResponse([
-      "supervisor" => $this->groupViewFactory->getGroups($user->getGroupsAsSupervisor()->getValues()),
-      "student" => $this->groupViewFactory->getGroups($user->getGroupsAsStudent()->getValues()),
+      "supervisor" => $this->groupViewFactory->getGroups($asSupervisor->getValues()),
+      "student" => $this->groupViewFactory->getGroups($asStudent->getValues()),
       "stats" => $user->getGroupsAsStudent()->map(
         function (Group $group) use ($user) {
           return $this->groupViewFactory->getStudentsStats($group, $user);
         }
-      )->getValues()
+      )->getValues(),
+      "archived" => [
```

Comment:
Someone, who uses ReCodEx for a long time, can get quite a lot (not always necessary) data here.